core/domain/user_services.py

```diff
@@ -1006,12 +971,9 @@ def get_human_readable_user_ids(user_ids):
             raise Exception('User not found.')
         elif user_settings.user_id == feconf.SYSTEM_COMMITTER_ID:
             usernames.append('admin')
-        elif user_settings.username:
-            usernames.append(user_settings.username)
         else:
-            usernames.append(
-                '[Awaiting user registration: %s]' %
-                user_settings.truncated_email)
+            usernames.append(user_settings.username)
```

Comment:
I thought I suggested raising an Exception if a user model has no username.

That said, such entities do exist in production. So, alternatively, you can keep the previous code, but you would need to create the user model manually in tests.

Reply:
I am sorry but I don't understand what you mean here. Did you see my comment [here](https://github.com/oppia/oppia/pull/6980#discussion_r297108666)? Should I add a check for empty username here:
https://github.com/oppia/oppia/blob/ee0e7fc732e82655122514c13101d23d5c1394e4/core/domain/user_services.py#L157

Reply:
> That said, such entities do exist in production. So, alternatively, you can keep the previous code, but you would need to create the user model manually in tests.

Done!